src/apps/chifra/pkg/cache/read.go

```diff
@@ -606,8 +607,11 @@ func ReadParameter(reader *bufio.Reader) (param *types.SimpleParameter, err erro
 		return
 	}
 
-	err = readString(reader, &param.Value)
-	if err != nil {
+	var jsonValue string
```

Comment:
Just curious. Does the C++ code store the articulated parameter as a JSON string?

Reply:
Not really, but it stores `Value` as string, always. 
This is a shortcut to read the value as string and then convert to the proper type. This makes sure that we output cached parameter's value just like non-cached one (the new code allows `Value` to be anything to make arrays first-class citizens when outputting JSON).

And, because `"some string value"` is valid JSON, it doesn't break when some simple type is cached

Reply:
Isn't this reading binary code? If the JSON object is not stored as a string, wouldn't this crash? Just curious. Maybe I'm misunderstanding.